### ISSUE TITLE
flaky tests: TestRPC_TLS_ are still unhappy

### ISSUE THREAD

**Issue by shoenig** [hashicorp/nomad#16253]
run from today: https://github.com/hashicorp/nomad/actions/runs/4255942116/jobs/7404222153

e.g.
```
=== FAIL: nomad TestRPC_TLS_Enforcement_RPC/other_region_client/servers_only_rpc/nomad_RPC:_rpc=Plan.Submit_verify_hostname=true_leader=false (0.00s)
    rpc_test.go:1329: 
        	Error Trace:	/home/runner/work/nomad/nomad/nomad/rpc_test.go:1329
        	Error:      	"write tcp 127.0.0.1:48888->127.0.0.1:43313: write: broken pipe" does not contain "certificate"
        	Test:       	TestRPC_TLS_Enforcement_RPC/other_region_client/servers_only_rpc/nomad_RPC:_rpc=Plan.Submit_verify_hostname=true_leader=false
```

**Comment by tgross** [comment]
I've got https://github.com/hashicorp/nomad/pull/18155 open with a possible fix. The tl;dr is that I can finally reproduce the error locally by setting `RPCHandshakeTimeout` very low. That's at least a _plausible_ cause in the high-contention environment that is GHA. But can't tell for certain without more debugging. I'm not going to close this even if that's merged until it's been a little since we've seen the error again.

**Comment by tgross** [comment]
Closed by #18155
